pkg/builder/local_build_executor.go

```diff
@@ -240,8 +271,12 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 	// Build rules generally expect the parent directories to already be
 	// there. We later use the directory handles to extract output files.
 	outputParentDirectories := map[string]filesystem.Directory{}
+	workingDirectory := command.WorkingDirectory
+	if workingDirectory == "" {
+		workingDirectory = "."
+	}
 	for _, outputDirectory := range command.OutputDirectories {
-		dirPath := path.Dir(outputDirectory)
+		dirPath := newOutputDirectory(workingDirectory, path.Dir(outputDirectory)).Path()
 		if _, ok := outputParentDirectories[dirPath]; !ok {
 			dir, err := be.createOutputParentDirectory(buildDirectory, dirPath)
```

Comment:
What if you were to leave all of the existing bookkeeping intact (so still keep `outputParentDirectories` indexed by the path relative to the working directory), but only adjusted this call to `createOutputParentDirectory()` to use the joined path? Wouldn't that work?

Reply:
So to clarify: revert tall of the changes to this source file, but replace this specific line with:
```go
dir, err := be.createOutputParentDirectory(buildDirectory, path.Join(command.WorkingDirectory, dirPath))
```